include documentation about the httpOption annotation (#4273)
knative/docs: docs/developer/serving/services/http-option.md

# Configuring HTTP(S) Redirect

> Operators can disable HTTP traffic by modifying the value of `httpProtocol` in the `config-network` ConfigMap. This is located in the `knative-serving` namespace and is part of Knative Serving's installation.

Author: I cut a lot of this section and linked to the appropriate operator page now

The available values are:

- `Enabled` - Services will accept HTTP traffic

> - `Redirected` - Services will send a 301 redirect for all HTTP connections, asking the clients to use HTTPS

Reviewer: ```suggestion
- `Redirected` - Services send a 301 redirect for all HTTP connections and ask the clients to use HTTPS instead
```

> The value set by the operator will be the default for all Knative Services.

Reviewer: ```suggestion
The value set by the Operator is the default setting for all Knative Services.
```

### Overriding the default HTTP behaviour

> Developers can override this default behavior for each Service by specifying the `networking.knative.dev/httpOption` annotation.

Reviewer: ```suggestion
You can override the default behavior for each Service by configuring the `networking.knative.dev/httpOption` annotation:
```
